### PR TITLE
added error logging for process_phonon_dos_YPHON

### PR DESCRIPTION
**Pull request by lukeamyers**
If some folders are complete, but others not. `process_phonon_dos_YPHON` will run for the complete ones, and log the incomplete one in the .log file.

### CODE REVIEW

File: dfttk/workflows.py [hunk 1]
```diff
@@ -679,6 +680,14 @@ def process_phonon_dos_YPHON(path: str):
         path (str): path to the folder containing all the phonon calculation folders. E.g. phonon_1, phonon_2, etc.
     """
 
+    # Configure logging
+    log_file_path = os.path.join(path, 'phonons_parallel.log')
```
Comment (nhew1994) [inline dfttk/workflows.py:688]: Maybe change to log_file_path = os.path.join(path, 'process_phonon_dos_YPHON.log') to reflect the function.

**Review by nhew1994 — COMMENTED** [review]
Good. But I need some help with the log_file_path. I want to implement some of the things below, but I don't know how at the moment:

- Every time this function is run, overwrite the entire log_file_path. At the moment, it appends to it. 
- If I delete the log_file_path and rerun the function, it will work, but the log_file_path will not be written.

**Review by nhew1994 — APPROVED** [review]

File: dfttk/workflows.py (hunk 1 above)
Comment (lukeamyers) [reply dfttk/workflows.py:688]: sure.

**Comment by lukeamyers** [comment]
>     * Every time this function is run, overwrite the entire log_file_path. At the moment, it appends to it.

before '# configure logging' check if the file exists. if it does delete it.

**Comment by lukeamyers** [comment]
> > ```
> > * Every time this function is run, overwrite the entire log_file_path. At the moment, it appends to it.
> > ```
> 
> before '# configure logging' check if the file exists. if it does delete it.

or perhaps just under the comment

**Comment by lukeamyers** [comment]
If it would be much easier for me to do it I will.

**Comment by lukeamyers** [comment]
We will likely need to get familiar with logging when we move to doing high-throughput workflows. So we might want to think about a general workflow log file or something of that sort.

File: dfttk/workflows.py (hunk 1 above)
Comment (nhew1994) [reply dfttk/workflows.py:688]: The issue is resolved for me if I use this in the beginning of the function:

# Reset logging configuration
    for handler in logging.root.handlers[:]:
        logging.root.removeHandler(handler)
        
    # Configure logging
    log_file_path = os.path.join(path, 'phonons_parallel.log')
    
    # If log_file_path already exists, delete it
    if os.path.exists(log_file_path):
        os.remove(log_file_path)
        
    logging.basicConfig(
        filename=log_file_path,
        level=logging.ERROR,  # Set the log level to ERROR
        format='%(asctime)s - %(levelname)s - %(message)s'
    )

Comment (nhew1994) [reply dfttk/workflows.py:688]: # And delete the empty file at the end

# Delete log_file_path if it is empty
        if os.path.exists(log_file_path) and os.path.getsize(log_file_path) == 0:
            os.remove(log_file_path)

**Comment by nhew1994** [comment]
You can merge when ready.

**Comment by lukeamyers** [comment]
> You can merge when ready.

Do you want to commit the changes you made locally? if it's easier I'll copy them from here then commit and merge

**Review by nhew1994 — COMMENTED** [review]
`def process_phonon_dos_YPHON(path: str):
    """Processes the phonon DOS calculations using YPHON.

    Args:
        path (str): path to the folder containing all the phonon calculation folders. E.g. phonon_1, phonon_2, etc.
    """

    # Reset logging configuration
    for handler in logging.root.handlers[:]:
        logging.root.removeHandler(handler)
        
    # Configure logging
    log_file_path = os.path.join(path, 'phonons_parallel.log')
    
    # If log_file_path already exists, delete it
    if os.path.exists(log_file_path):
        os.remove(log_file_path)
        
    logging.basicConfig(
        filename=log_file_path,
        level=logging.ERROR,  # Set the log level to ERROR
        format='%(asctime)s - %(levelname)s - %(message)s'
    )
    
    # Go to each phonon folder and copy the CONTCAR, OUTCAR, and vasprun.xml files to the phonon_dos folder to be processed by YPHON
    phonon_folders = [
        folder
        for folder in os.listdir(path)
        if os.path.isdir(os.path.join(path, folder)) and folder.startswith("phonon")
    ]

    for phonon_folder in phonon_folders:
        try:
            phonon_dos_folder = os.path.join(path, phonon_folder, "phonon_dos")
            phonon_folder = os.path.join(path, phonon_folder)
            if not os.path.exists(phonon_dos_folder):
                os.makedirs(phonon_dos_folder, exist_ok=True)
            shutil.copy(
                os.path.join(phonon_folder, "CONTCAR.2phonons"),
                os.path.join(phonon_dos_folder, "CONTCAR"),
            )
            shutil.copy(
                os.path.join(phonon_folder, "OUTCAR.2phonons"),
                os.path.join(phonon_dos_folder, "OUTCAR"),
            )
            shutil.copy(
                os.path.join(phonon_folder, "vasprun.xml.2phonons"),
                os.path.join(phonon_dos_folder, "vasprun.xml"),
            )

            index = phonon_folder.split("_")[-1]
            structure = Structure.from_file(os.path.join(phonon_dos_folder, "CONTCAR"))
            number_of_atoms = structure.num_sites
            volume = extract_volume(os.path.join(phonon_dos_folder, "CONTCAR"))
            volume_per_atom = volume / number_of_atoms

            with open(os.path.join(phonon_dos_folder, "volph_" + index), "w") as f:
                f.write(str(volume_per_atom))

            # YPHON commands
            subprocess.run(["vasp_fij"], cwd=phonon_dos_folder)
            subprocess.run(["Yphon <superfij.out"], cwd=phonon_dos_folder, shell=True)

            os.rename(
                os.path.join(phonon_dos_folder, "vdos.out"),
                os.path.join(phonon_dos_folder, "vdos_" + index),
            )
        except Exception as e:
            logging.error(f"Error processing folder {phonon_folder}: {e}")

    os.makedirs(os.path.join(path, "YPHON_results"), exist_ok=True)
    for phonon_folder in phonon_folders:
        try:
            phonon_dos_folder = os.path.join(path, phonon_folder, "phonon_dos")
            phonon_folder = os.path.join(path, phonon_folder)
            index = phonon_folder.split("_")[-1]
            shutil.copy(
                os.path.join(phonon_dos_folder, "vdos_" + index),
                os.path.join(path, "YPHON_results", "vdos_" + index),
            )
            shutil.copy(
                os.path.join(phonon_dos_folder, "volph_" + index),
                os.path.join(path, "YPHON_results", "volph_" + index),
            )
        except Exception as e:
            logging.error(f"Error copying files from {phonon_folder}: {e}")

        # Delete log_file_path if it is empty
        if os.path.exists(log_file_path) and os.path.getsize(log_file_path) == 0:
            os.remove(log_file_path)

The function I have is this. Think it is easier if you copy this and commit. `

**Pull request merged**